infra/modules/monitoring/main.tf

```diff
@@ -0,0 +1,126 @@
+  ok_actions                = [aws_sns_topic.this.arn]
+  insufficient_data_actions = [aws_sns_topic.this.arn]
+  dimensions = {
+    ClusterName = var.cluster_name
+    ServiceName = var.service_name
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "task_health_alarm" {
+  count                     = var.task_health_percentage_threshold != null ? 1 : 0
+  alarm_name                = "${var.cluster_name}-task-health"
+  comparison_operator       = "LessThanThreshold"
+  evaluation_periods        = 1
+  metric_name               = "TaskHealthPercentage"
+  namespace                 = "AWS/ECS"
+  period                    = 60
+  statistic                 = "Minimum"
+  threshold                 = var.task_health_percentage_threshold
+  alarm_description         = "Alarm triggered when any ECS task becomes unhealthy."
+  alarm_actions             = [aws_sns_topic.this.arn]
+  ok_actions                = [aws_sns_topic.this.arn]
+  insufficient_data_actions = [aws_sns_topic.this.arn]
+  dimensions = {
+    ClusterName = var.cluster_name
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "task_placement_errors_alarm" {
+  count                     = var.task_placement_error_threshold != null ? 1 : 0
+  alarm_name                = "${var.cluster_name}-task-placement-errors"
+  comparison_operator       = "GreaterThanThreshold"
+  evaluation_periods        = 3
+  metric_name               = "TaskPlacementError"
+  namespace                 = "AWS/ECS"
+  period                    = 300
+  statistic                 = "Sum"
+  threshold                 = var.task_placement_error_threshold
+  alarm_description         = "Alarm triggered when ECS tasks encounter placement errors."
+  alarm_actions             = [aws_sns_topic.this.arn]
+  ok_actions                = [aws_sns_topic.this.arn]
+  insufficient_data_actions = [aws_sns_topic.this.arn]
+  dimensions = {
+    ClusterName = var.cluster_name
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "service_availability_alarm" {
+  count                     = var.service_availability_threshold != null ? 1 : 0
+  alarm_name                = "${var.cluster_name}-service-availability"
+  comparison_operator       = "LessThanThreshold"
+  evaluation_periods        = 1
+  metric_name               = "ServiceAvailability"
+  namespace                 = "AWS/ECS"
+  period                    = 60
+  statistic                 = "Minimum"
+  threshold                 = var.service_availability_threshold
+  alarm_description         = "Alarm triggered when an ECS service becomes unavailable."
+  alarm_actions             = [aws_sns_topic.this.arn]
+  ok_actions                = [aws_sns_topic.this.arn]
+  insufficient_data_actions = [aws_sns_topic.this.arn]
+  dimensions = {
+    ClusterName = var.cluster_name
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "network_connectivity_alarm" {
+  count                     = var.high_network_packet_loss_threshold != null ? 1 : 0
```

Comment:
what conditions would trigger this alarm, and what is an engineer supposed to do if there's network connectivity issues within ECS? if it's not actionable then we should probably get rid of the alarm

Reply:
as we agreed those alerts were removed.